### ISSUE TITLE
OverflowError: int too large to convert to float

### ISSUE THREAD

**Issue by haik** [n1analytics/python-paillier#62]
When I try to decrypt the ciphertext by using _priv.decrypt(n1_r1_enc)_ and n1_r1_enc is the sum of n1 and r1, it raised the following error.
> File "/usr/local/lib/python3.6/site-packages/phe/paillier.py", line 287, in decrypt
    return encoded.decode()
   File "/usr/local/lib/python3.6/site-packages/phe/encoding.py", line 220, in decode
    return mantissa * pow(self.BASE, self.exponent)

It is weird because it can decrypt correctly when n1 and r1 are the following values.
> r1: 462
   n1: 2.954413649165506
   n1+r1: 464.95441364916553

The error occurred when n1 and r1 are the following values:
> r1: 593
   n1: 0.1641340916202469

or:
> r1: 445
   n1: 0.16413409062205825

I really have no idea about what is going wrong.

**Comment by gusmith** [comment]
I tried to repeat the issue, but I was not able to.
Could you specify:
 - how you created your key pair
 - how you encrypted the numbers
 - how you summed the numbers together
 - if gmpy2 is installed?

See the branch [issue-62](https://github.com/n1analytics/python-paillier/tree/issue-62) d7e6e58011d7f49b5e65e38440ec631b6535ffe7 where I added a test which repeat what you are describing in this issue, but I did not get any exceptions.

**Comment by nbgl** [comment]
Here’s how I can reproduce it:
```python
>>> pub, priv = phe.generate_paillier_keypair()
>>> a = pub.encrypt(445)
>>> b = pub.encrypt(0.16413409062205825) + pub.encrypt(2 ** -965)  # Force big exponent.
>>> priv.decrypt(a + b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jakub/Repositories/python-paillier/phe/paillier.py", line 290, in decrypt
    return encoded.decode()
  File "/Users/jakub/Repositories/python-paillier/phe/encoding.py", line 220, in decode
    return mantissa * pow(self.BASE, self.exponent)
OverflowError: int too large to convert to float
```

This issue is fixed with #73.
